sklearn/preprocessing/tests/test_polynomial.py

```diff
@@ -788,6 +793,207 @@ def test_polynomial_features_csr_X_dim_edges(deg, dim, interaction_only):
     assert_array_almost_equal(Xt_csr.A, Xt_dense)
 
 
+def test_csr_polynomial_expansion_index_overflow_non_regression():
```

Comment:
I think this test must be at least parametrized on `interaction_only` and potentially on `include_bias`.

What do you think?